src/Mod/Path/PathScripts/drillableLib.py

```diff
@@ -0,0 +1,235 @@
+import PathScripts.PathLog as PathLog
+import FreeCAD as App
+import Part
+import numpy
+import math
+
+if False:
+    PathLog.setLevel(PathLog.Level.DEBUG, PathLog.thisModule())
+    PathLog.trackModule(PathLog.thisModule())
+else:
+    PathLog.setLevel(PathLog.Level.INFO, PathLog.thisModule())
+
+def isDrillableCylinder(obj, candidate, tooldiameter=None, vector=App.Vector(0, 0, 1)):
+    """
+    checks if a candidate cylindrical face is drillable
+    """
+
+    matchToolDiameter = tooldiameter is not None
+    matchVector = vector is not None
+
+    PathLog.debug(
+        "\n match tool diameter {} \n match vector {}".format(
+            matchToolDiameter, matchVector
+        )
+    )
+
+    def raisedFeature(obj, candidate):
+        # check if the cylindrical 'lids' are inside the base
+        # object.  This eliminates extruded circles but allows
+        # actual holes.
+
+        startLidCenter = App.Vector(
+            candidate.BoundBox.Center.x,
+            candidate.BoundBox.Center.y,
+            candidate.BoundBox.ZMax,
+        )
+
+        endLidCenter = App.Vector(
+            candidate.BoundBox.Center.x,
+            candidate.BoundBox.Center.y,
+            candidate.BoundBox.ZMin,
+        )
+
+        return obj.isInside(startLidCenter, 1e-6, False) or obj.isInside(
+            endLidCenter, 1e-6, False
+        )
+
+    def getSeam(candidate):
+        # Finds the vertical seam edge in a cylinder
+
+        for e in candidate.Edges:
+            if isinstance(e.Curve, Part.Line):  # found the seam
+                return e
+
+    if not candidate.ShapeType == "Face":
+        raise TypeError("expected a Face")
+
+    if not str(candidate.Surface) == "<Cylinder object>":
```

Comment:
Is there a reason for converting to string here?
Would `isinstance(candidate.Surface, Part.Cylinder)` work as well? Since isinstance is used on line 52.

Reply:
I think you're correct.  I'll change it to use isInstance().   I think it's a bit of copy pasta left.